backward-compatibility/generate-v0.13.0/src/data_0_13.rs

```diff
@@ -0,0 +1,1209 @@
+        store_versioned_test!(&pub_data_type, dir, &PUB_DATA_TYPE.test_filename);
+
+        TestMetadataKmsGrpc::PubDataType(PUB_DATA_TYPE)
+    }
+
+    fn gen_priv_data_type(dir: &PathBuf) -> TestMetadataKmsGrpc {
+        let priv_data_type = PrivDataType::ContextInfo;
+        store_versioned_test!(&priv_data_type, dir, &PRIV_DATA_TYPE.test_filename);
+
+        TestMetadataKmsGrpc::PrivDataType(PRIV_DATA_TYPE)
+    }
+}
+
+impl KMSCoreVersion for V0_13 {
+    const VERSION_NUMBER: &'static str = "0.12.2";
+
+    // Without this, some keys will be generated differently every time we run the script
+    fn seed_prng(seed: u128) {
+        let mut seeder = DeterministicSeeder::<DefaultRandomGenerator>::new(Seed(seed));
+        let shortint_engine = ShortintEngine::new_from_seeder(&mut seeder);
+        ShortintEngine::with_thread_local_mut(|local_engine| {
+            let _ = std::mem::replace(local_engine, shortint_engine);
+        });
+    }
+
+    fn gen_kms_data() -> Vec<TestMetadataKMS> {
+        let dir = Self::data_dir().join(KMS_MODULE_NAME);
+        create_dir_all(&dir).unwrap();
+
+        vec![
+            KmsV0_13::gen_private_sig_key(&dir),
+            KmsV0_13::gen_public_sig_key(&dir),
+            KmsV0_13::gen_app_key_blob(&dir),
+            KmsV0_13::gen_key_gen_metadata(&dir),
+            KmsV0_13::gen_typed_plaintext(&dir),
+            KmsV0_13::gen_signcryption_payload(&dir),
+            KmsV0_13::gen_signcryption_key(&dir),
+            KmsV0_13::gen_designcryption_key(&dir),
+            KmsV0_13::gen_unified_signcryption(&dir),
+            KmsV0_13::gen_backup_ciphertext(&dir),
+            KmsV0_13::gen_unified_cipher(&dir),
+            KmsV0_13::gen_hybrid_kem_ct(&dir),
+            KmsV0_13::gen_recovery_material(&dir),
+            KmsV0_13::gen_internal_cus_context_handles(&dir),
+            KmsV0_13::gen_internal_cus_setup_msg(&dir),
+            KmsV0_13::gen_kms_fhe_key_handles(&dir),
+            KmsV0_13::gen_threshold_fhe_keys(&dir),
+            KmsV0_13::gen_internal_cus_rec_out(&dir),
+            KmsV0_13::gen_operator_backup_output(&dir),
+        ]
+    }
+
+    fn gen_threshold_fhe_data() -> Vec<TestMetadataDD> {
+        let dir = Self::data_dir().join(DISTRIBUTED_DECRYPTION_MODULE_NAME);
+        create_dir_all(&dir).unwrap();
+
+        vec![
+            DistributedDecryptionV0_13::gen_prss_setup_rpoly_64(&dir),
+            DistributedDecryptionV0_13::gen_prss_setup_rpoly_128(&dir),
+            DistributedDecryptionV0_13::gen_prf_key(&dir),
+        ]
+    }
+
+    fn gen_kms_grpc_data() -> Vec<TestMetadataKmsGrpc> {
+        let dir = Self::data_dir().join(KMS_GRPC_MODULE_NAME);
+        create_dir_all(&dir).unwrap();
+
+        vec![
+            KmsGrpcV0_13::gen_signed_pub_data_handle_internal(&dir),
+            KmsGrpcV0_13::gen_public_key_type(&dir),
+            KmsGrpcV0_13::gen_pub_data_type(&dir),
+            KmsGrpcV0_13::gen_priv_data_type(&dir),
+        ]
+    }
+}
```

Comment:
I don't think we need to always add everything to every data generation binary. only the new data that got changed in this version (in this case 0.13) need to be added. otherwise we'll have a lot of duplicate code.

at least this is how it's managed on the tfhe-rs side.